tests/torch/test_compile_torch.py

```diff
@@ -1373,3 +1373,32 @@ def test_mono_parameter_rounding_warning(
             verbose=False,
             get_and_compile=False,
         )
+
+
+def test_onnx_no_input():
+    """Test a torch model that has no input when converted to onnx."""
+
+    torch_input = torch.randn(100, 28)
+
+    class SimplifiedNet(nn.Module):
+        """Network with no input in the onnx graph."""
+
+        def __init__(self):
+            super().__init__()
+            self.fc1 = nn.Linear(28, 10)
+            self.threshold = torch.tensor(0.5, dtype=torch.float32)
+
+        def forward(self, x):
+            """Forward pass."""
+            zeros = numpy.zeros_like(x)
+            x = x + zeros
+            x = (x > self.threshold).to(torch.float32)
+            x = self.fc1(x)
+            return x
+
+    model = SimplifiedNet()
+
+    with pytest.raises(AssertionError) as excinfo:
```

Comment:
as said above, this should be a ValueError then !

Reply:
Oh yeah I forgot about this. Will fix.

Reply:
Should be good.

Reply:
actually you were right not to change anything here this was a legit assert my bad

Reply:
not sure I understood? I actually think you were right? Why not using the feature from pytest.raises instead of creating a new line with an assert?